var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -30,15 +30,21 @@ class Ipopt(AutotoolsPackage):
             description="Build with METIS partitioning support")
     variant('debug', default=False,
             description="Build debug instead of optimized version")
+    variant('mumps', default=True,
+            description='Build with support for linear solver MUMPS')
 
     depends_on("blas")
     depends_on("lapack")
     depends_on("pkgconfig", type='build')
-    depends_on("mumps+double~mpi")
+    depends_on("mumps+double~mpi", when='+mumps')
     depends_on('coinhsl', when='+coinhsl')
     depends_on('metis@4.0:', when='+metis')
 
+    # Must have at least one linear solver available!
+    conflicts('~mumps', when='~coinhsl')
+
     patch('ipopt_ppc_build.patch', when='arch=ppc64le')
+    patch('windows_clang_compiler_conflict.patch')
```

Comment:
The patch doesn't apply to all versions, can you restrict it to those where it does?